### ISSUE TITLE
bug: http get broken

### ISSUE THREAD

**Issue by jcesarmobile** [ionic-team/capacitor#6751]
### Capacitor Version
main branch

### Platform(s)
ios
android

### Current Behavior

While triaging https://github.com/ionic-team/capacitor/issues/6645, I updated the provided app to use the code from main branch as it supposedly fixes formData, but it now fails in the previous step where it was using a get call to an image to be posted later.

```
ERROR
HttpErrorResponse
error: null
headers: HttpHeaders {normalizedNames: Map, lazyUpdate: null, lazyInit: function}
message: "Http failure response for assets/example-file.png: 0 OK"
name: "HttpErrorResponse"
ok: false
status: 0
statusText: "OK"
url: "assets/example-file.png"
```

### Expected Behavior

get doesn't fail

### Code Reproduction
use this app and update it to use the code from main branch of core/ios/android
https://github.com/dennisameling/capacitor-http-bug

**Comment by ghenry22** [comment]
may well be related to https://github.com/ionic-team/capacitor/issues/6739

**Comment by koraxos** [comment]
Hello, a fix has been merged for capacitor 4.8.1 https://github.com/ionic-team/capacitor/commit/3cd7196ec10066a98561e5d1b1166d8c1133f13a , but i'm not sure that it has been merged on the main branch (cap 5 )
